staging/src/k8s.io/pod-security-admission/admission/admission.go

```diff
@@ -407,7 +407,10 @@ func (a *Admission) EvaluatePod(ctx context.Context, nsPolicy api.Policy, nsPoli
 		klog.V(2).InfoS("failed to parse PodSecurity namespace labels", "err", nsPolicyErr)
 		auditAnnotations["error"] = fmt.Sprintf("Failed to parse policy: %v", nsPolicyErr)
 	}
-	// TODO: log nsPolicy evaluation with context (op, resource, namespace, name) for the request.
+
+	if klog.V(4).Enabled() {
+		klog.InfoS("Pod Security evaluation", "policy", fmt.Sprintf("%v", nsPolicy), "op", attrs.GetOperation(), "resource", attrs.GetResource(), "namespace", attrs.GetNamespace(), "name", attrs.GetName())
```

Comment:
does this log N times for a single update API call on a namespace containing N pods? I don't think we want to log every individual pod checked in an update namespace request.

Reply:
Oh, maybe we can just log it when evaluate failed?

Reply:
@tallclair WDYT🙈

Reply:
I'd expect at most one log line for a namespace update. That means we should probably log one function up, not in this low level method

Reply:
No, `EvaluatePod` isn't called on namespace evaluations (that's `EvaluatePodsInNamespace`). I like logging here after all exemptions have been evaluated (so this means the pod was actually evaluated).

Reply:
ah, missed that